python/lsst/pipe/tasks/skyCorrection.py

```diff
@@ -183,64 +106,100 @@ class SkyCorrectionConnections(pipeBase.PipelineTaskConnections, dimensions=("in
+        doc="Calibration sky frames.",
         name="sky",
         multiple=True,
         storageClass="ExposureF",
         dimensions=["instrument", "physical_filter", "detector"],
         isCalibration=True,
-        lookupFunction=_skyLookup,
+        lookupFunction=_skyFrameLookup,
     )
-    calExpCamera = cT.Output(
-        doc="Output camera image.",
-        name='calexp_camera',
-        storageClass="ImageF",
-        dimensions=["instrument", "visit"],
+    camera = cT.PrerequisiteInput(
+        doc="Input camera.",
+        name="camera",
+        storageClass="Camera",
+        dimensions=["instrument"],
+        isCalibration=True,
     )
     skyCorr = cT.Output(
-        doc="Output sky corrected images.",
-        name='skyCorr',
+        doc="Sky correction data, to be subtracted from the calibrated exposures.",
+        name="skyCorr",
         multiple=True,
         storageClass="Background",
         dimensions=["instrument", "visit", "detector"],
     )
+    calExpMosaic = cT.Output(
+        doc="Full focal plane mosaicked image of the sky corrected calibrated exposures.",
+        name="calexp_skyCorr_visit_mosaic",
+        storageClass="ImageF",
+        dimensions=["instrument", "visit"],
+    )
+    calBkgMosaic = cT.Output(
+        doc="Full focal plane mosaicked image of the sky corrected calibrated exposure backgrounds.",
+        name="calexpBackground_skyCorr_visit_mosaic",
+        storageClass="ImageF",
+        dimensions=["instrument", "visit"],
+    )
 
 
-class SkyCorrectionConfig(pipeBase.PipelineTaskConfig, pipelineConnections=SkyCorrectionConnections):
-    """Configuration for SkyCorrectionTask"""
-    bgModel = ConfigField(dtype=FocalPlaneBackgroundConfig, doc="Background model")
-    bgModel2 = ConfigField(dtype=FocalPlaneBackgroundConfig, doc="2nd Background model")
-    sky = ConfigurableField(target=SkyMeasurementTask, doc="Sky measurement")
-    maskObjects = ConfigurableField(target=MaskObjectsTask, doc="Mask Objects")
-    doMaskObjects = Field(dtype=bool, default=True, doc="Mask objects to find good sky?")
-    doBgModel = Field(dtype=bool, default=True, doc="Do background model subtraction?")
-    doBgModel2 = Field(dtype=bool, default=True, doc="Do cleanup background model subtraction?")
-    doSky = Field(dtype=bool, default=True, doc="Do sky frame subtraction?")
-    binning = Field(dtype=int, default=8, doc="Binning factor for constructing focal-plane images")
-    calexpType = Field(dtype=str, default="calexp",
-                       doc="Should be set to fakes_calexp if you want to process calexps with fakes in.")
+class SkyCorrectionConfig(PipelineTaskConfig, pipelineConnections=SkyCorrectionConnections):
+    maskObjects = ConfigurableField(
+        target=MaskObjectsTask,
+        doc="Mask Objects",
+    )
+    doMaskObjects = Field(
+        dtype=bool,
+        default=True,
+        doc="Iteratively mask objects to find good sky?",
+    )
+    bgModel = ConfigField(
```

Comment:
If you are going to introduce `bgModel1`, then this should be deprecated and set `optional=True`.

Reply:
As discussed via Slack, there is no `optional` argument for a `ConfigField`. Instead, I have set the `dtype` for this `ConfigField` to the base `Config` class (to allow `validate` to pass successfully) and left comments informing the user that this field should not be used - please use `bgModel1` instead.